cmd/validate.go

```diff
@@ -87,136 +87,141 @@ func invokeDigraphValidateAPI(parsedTFPlan utils.ParsedTerraformPlan, digraphAPI
 	defer res.Body.Close()
 	body, _ := io.ReadAll(res.Body)
 
+	if res.StatusCode != http.StatusOK {
+		return "", errors.New(fmt.Sprintf("client: error with http request: status code %d with body %s\n", res.StatusCode, body))
+	}
+
 	return string(body), nil
 }
 
-func validate() *cobra.Command {
-	cmd := &cobra.Command{
-		Use:       "validate",
-		Short:     "Validate infra config changes",
-		Long:      ``,
-		ValidArgs: []string{"--", "-"},
-		RunE: func(cmd *cobra.Command, args []string) error {
-			tfPlanOutput, _ := cmd.Flags().GetString("raw-output-plan")
-			jsonPathPlan, _ := cmd.Flags().GetString("json-path-plan")
-			tfRawPath, _ := cmd.Flags().GetString("output-path-plan")
-			tfJsonOutput, _ := cmd.Flags().GetString("json-output-plan")
-
-			terraformAPIKey, _ := cmd.Flags().GetString("terraform-api-key")
-			digraphAPIKey, _ := cmd.Flags().GetString("api-key")
-			repository, _ := cmd.Flags().GetString("repository")
-			ref, _ := cmd.Flags().GetString("ref")
-
-			issueNumber, _ := cmd.Flags().GetInt("issue-number")
-			commitSHA, _ := cmd.Flags().GetString("commit-sha")
-
-			mode, _ := cmd.Flags().GetString("mode")
-			terraformWorkspace, _ := cmd.Flags().GetString("terraform-workspace")
-
-			if len(digraphAPIKey) == 0 {
-				err := godotenv.Load(".env")
+var validate = &cobra.Command{
+	Use:       "validate",
+	Short:     "Validate infra config changes",
+	Long:      ``,
+	ValidArgs: []string{"--", "-"},
+	RunE: func(cmd *cobra.Command, args []string) error {
+		tfPlanOutput, _ := cmd.Flags().GetString("raw-output-plan")
+		jsonPathPlan, _ := cmd.Flags().GetString("json-path-plan")
+		tfRawPath, _ := cmd.Flags().GetString("output-path-plan")
+		tfJsonOutput, _ := cmd.Flags().GetString("json-output-plan")
 
-				if err != nil {
-					return fmt.Errorf("must specify api-key as argument or set it within a .env file")
-				}
+		terraformAPIKey, _ := cmd.Flags().GetString("terraform-api-key")
+		digraphAPIKey, _ := cmd.Flags().GetString("api-key")
+		repository, _ := cmd.Flags().GetString("repository")
+		ref, _ := cmd.Flags().GetString("ref")
 
-				digraphAPIKey = os.Getenv("api-key")
-			}
+		issueNumber, _ := cmd.Flags().GetInt("issue-number")
```

Comment:
for this getInt, if it isn't an int, does it throw an error? i would assume so, and we might want to capture that and handle it gracefully?

Reply:
the cli setup handles this- it will throw an error and the command will execute 
<img width="635" alt="image" src="https://user-images.githubusercontent.com/22775855/202278688-9e14cd28-93de-486d-9c6e-9df3bd8c8742.png">